// nem2-library/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});
exports.subnamespaceNamespaceId = exports.subnamespaceParentId = exports.namespaceId = exports.mosaicId = exports.address = exports.uint64 = exports.IdGenerator = exports.sha3Hasher = exports.convert = exports.nacl_catapult = exports.crypto = exports.KeyPair = exports.VerifiableTransactionBuilder = exports.VerifiableTransaction = exports.TransferTransaction = exports.SecretProofTransaction = exports.SecretLockTransaction = exports.NamespaceCreationTransaction = exports.MultisigModificationTransaction = exports.MosaicSupplyChangeTransaction = exports.MosaicCreationTransaction = exports.HashLockTransaction = exports.deadline = exports.AggregateTransaction = exports.CosignatureTransaction = exports.Schema = exports.UnconfirmedTransactionsListener = exports.TransactionStatusListener = exports.PartialTransactionsListener = exports.ConfirmedTransactionsListener = exports.BlockListener = exports.NetworkRoutesApi = exports.TransactionRoutesApi = exports.NamespaceRoutesApi = exports.MosaicRoutesApi = exports.BlockchainRoutesApi = exports.AccountRoutesApi = exports.UInt64DTO = exports.TransactionStatusDTO = exports.TransactionPayload = exports.TransactionIds = exports.TransactionHashes = exports.NamespaceNameDTO = exports.NamespaceMosaicMetaDTO = exports.NamespaceInfoDTO = exports.NamespaceIds = exports.NamespaceDTO = exports.MultisigAccountInfoDTO = exports.MultisigAccountGraphInfoDTO = exports.Multisig = exports.MosaicPropertiesDTO = exports.MosaicNameDTO = exports.MosaicInfoDTO = exports.MosaicIds = exports.MosaicDefinitionDTO = exports.MosaicDTO = exports.HeightDTO = exports.BlockchainStorageInfoDTO = exports.BlockchainScoreDTO = exports.BlockMetaDTO = exports.BlockInfoDTO = exports.BlockDTO = exports.AnnounceTransactionInfoDTO = exports.AccountMetaDTO = exports.AccountInfoDTO = exports.AccountIds = exports.AccountDTO = exports.ApiClient = undefined;

var _ApiClient = require('./ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _AccountDTO = require('./model/AccountDTO');

var _AccountDTO2 = _interopRequireDefault(_AccountDTO);

var _AccountIds = require('./model/AccountIds');

var _AccountIds2 = _interopRequireDefault(_AccountIds);

var _AccountInfoDTO = require('./model/AccountInfoDTO');

var _AccountInfoDTO2 = _interopRequireDefault(_AccountInfoDTO);

var _AccountMetaDTO = require('./model/AccountMetaDTO');

var _AccountMetaDTO2 = _interopRequireDefault(_AccountMetaDTO);

var _AnnounceTransactionInfoDTO = require('./model/AnnounceTransactionInfoDTO');

var _AnnounceTransactionInfoDTO2 = _interopRequireDefault(_AnnounceTransactionInfoDTO);

var _BlockDTO = require('./model/BlockDTO');

var _BlockDTO2 = _interopRequireDefault(_BlockDTO);

var _BlockInfoDTO = require('./model/BlockInfoDTO');

var _BlockInfoDTO2 = _interopRequireDefault(_BlockInfoDTO);

var _BlockMetaDTO = require('./model/BlockMetaDTO');

var _BlockMetaDTO2 = _interopRequireDefault(_BlockMetaDTO);

var _BlockchainScoreDTO = require('./model/BlockchainScoreDTO');

var _BlockchainScoreDTO2 = _interopRequireDefault(_BlockchainScoreDTO);

var _BlockchainStorageInfoDTO = require('./model/BlockchainStorageInfoDTO');

var _BlockchainStorageInfoDTO2 = _interopRequireDefault(_BlockchainStorageInfoDTO);

var _HeightDTO = require('./model/HeightDTO');

var _HeightDTO2 = _interopRequireDefault(_HeightDTO);

var _MosaicDTO = require('./model/MosaicDTO');

var _MosaicDTO2 = _interopRequireDefault(_MosaicDTO);

var _MosaicDefinitionDTO = require('./model/MosaicDefinitionDTO');

var _MosaicDefinitionDTO2 = _interopRequireDefault(_MosaicDefinitionDTO);

var _MosaicIds = require('./model/MosaicIds');

var _MosaicIds2 = _interopRequireDefault(_MosaicIds);

var _MosaicInfoDTO = require('./model/MosaicInfoDTO');

var _MosaicInfoDTO2 = _interopRequireDefault(_MosaicInfoDTO);

var _MosaicNameDTO = require('./model/MosaicNameDTO');

var _MosaicNameDTO2 = _interopRequireDefault(_MosaicNameDTO);

var _MosaicPropertiesDTO = require('./model/MosaicPropertiesDTO');

var _MosaicPropertiesDTO2 = _interopRequireDefault(_MosaicPropertiesDTO);

var _Multisig = require('./model/Multisig');

var _Multisig2 = _interopRequireDefault(_Multisig);

var _MultisigAccountGraphInfoDTO = require('./model/MultisigAccountGraphInfoDTO');

var _MultisigAccountGraphInfoDTO2 = _interopRequireDefault(_MultisigAccountGraphInfoDTO);

var _MultisigAccountInfoDTO = require('./model/MultisigAccountInfoDTO');

var _MultisigAccountInfoDTO2 = _interopRequireDefault(_MultisigAccountInfoDTO);

var _NamespaceDTO = require('./model/NamespaceDTO');

var _NamespaceDTO2 = _interopRequireDefault(_NamespaceDTO);

var _NamespaceIds = require('./model/NamespaceIds');

var _NamespaceIds2 = _interopRequireDefault(_NamespaceIds);

var _NamespaceInfoDTO = require('./model/NamespaceInfoDTO');

var _NamespaceInfoDTO2 = _interopRequireDefault(_NamespaceInfoDTO);

var _NamespaceMosaicMetaDTO = require('./model/NamespaceMosaicMetaDTO');

var _NamespaceMosaicMetaDTO2 = _interopRequireDefault(_NamespaceMosaicMetaDTO);

var _NamespaceNameDTO = require('./model/NamespaceNameDTO');

var _NamespaceNameDTO2 = _interopRequireDefault(_NamespaceNameDTO);

var _TransactionHashes = require('./model/TransactionHashes');

var _TransactionHashes2 = _interopRequireDefault(_TransactionHashes);

var _TransactionIds = require('./model/TransactionIds');

var _TransactionIds2 = _interopRequireDefault(_TransactionIds);

var _TransactionPayload = require('./model/TransactionPayload');

var _TransactionPayload2 = _interopRequireDefault(_TransactionPayload);

var _TransactionStatusDTO = require('./model/TransactionStatusDTO');

var _TransactionStatusDTO2 = _interopRequireDefault(_TransactionStatusDTO);

var _UInt64DTO = require('./model/UInt64DTO');

var _UInt64DTO2 = _interopRequireDefault(_UInt64DTO);

var _AccountRoutesApi = require('./api/AccountRoutesApi');

var _AccountRoutesApi2 = _interopRequireDefault(_AccountRoutesApi);

var _BlockchainRoutesApi = require('./api/BlockchainRoutesApi');

var _BlockchainRoutesApi2 = _interopRequireDefault(_BlockchainRoutesApi);

var _MosaicRoutesApi = require('./api/MosaicRoutesApi');

var _MosaicRoutesApi2 = _interopRequireDefault(_MosaicRoutesApi);

var _NamespaceRoutesApi = require('./api/NamespaceRoutesApi');

var _NamespaceRoutesApi2 = _interopRequireDefault(_NamespaceRoutesApi);

var _TransactionRoutesApi = require('./api/TransactionRoutesApi');

var _TransactionRoutesApi2 = _interopRequireDefault(_TransactionRoutesApi);

var _NetworkRoutesApi = require('./api/NetworkRoutesApi');

var _NetworkRoutesApi2 = _interopRequireDefault(_NetworkRoutesApi);

var _BlockListener = require('./listeners/BlockListener');

var _BlockListener2 = _interopRequireDefault(_BlockListener);

var _ConfirmedTransactionsListener = require('./listeners/ConfirmedTransactionsListener');

var _ConfirmedTransactionsListener2 = _interopRequireDefault(_ConfirmedTransactionsListener);

var _PartialTransactionsListener = require('./listeners/PartialTransactionsListener');

var _PartialTransactionsListener2 = _interopRequireDefault(_PartialTransactionsListener);

var _TransactionStatusListener = require('./listeners/TransactionStatusListener');

var _TransactionStatusListener2 = _interopRequireDefault(_TransactionStatusListener);

var _UnconfirmedTransactionsListener = require('./listeners/UnconfirmedTransactionsListener');

var _UnconfirmedTransactionsListener2 = _interopRequireDefault(_UnconfirmedTransactionsListener);

var _Schema = require('./schema/Schema');

var _Schema2 = _interopRequireDefault(_Schema);

var _CosignatureTransaction = require('./transactions/CosignatureTransaction');

var _CosignatureTransaction2 = _interopRequireDefault(_CosignatureTransaction);

var _AggregateTransaction = require('./transactions/AggregateTransaction');

var _AggregateTransaction2 = _interopRequireDefault(_AggregateTransaction);

var _HashLockTransaction = require('./transactions/HashLockTransaction');

var _HashLockTransaction2 = _interopRequireDefault(_HashLockTransaction);

var _Deadline = require('./transactions/Deadline');

var _Deadline2 = _interopRequireDefault(_Deadline);

var _MosaicCreationTransaction = require('./transactions/MosaicCreationTransaction');

var _MosaicCreationTransaction2 = _interopRequireDefault(_MosaicCreationTransaction);

var _MosaicSupplyChangeTransaction = require('./transactions/MosaicSupplyChangeTransaction');

var _MosaicSupplyChangeTransaction2 = _interopRequireDefault(_MosaicSupplyChangeTransaction);

var _MultisigModificationTransaction = require('./transactions/MultisigModificationTransaction');

var _MultisigModificationTransaction2 = _interopRequireDefault(_MultisigModificationTransaction);

var _NamespaceCreationTransaction = require('./transactions/NamespaceCreationTransaction');

var _NamespaceCreationTransaction2 = _interopRequireDefault(_NamespaceCreationTransaction);

var _SecretLockTransaction = require('./transactions/SecretLockTransaction');

var _SecretLockTransaction2 = _interopRequireDefault(_SecretLockTransaction);

var _SecretProofTransaction = require('./transactions/SecretProofTransaction');

var _SecretProofTransaction2 = _interopRequireDefault(_SecretProofTransaction);

var _TransferTransaction = require('./transactions/TransferTransaction');

var _TransferTransaction2 = _interopRequireDefault(_TransferTransaction);

var _VerifiableTransaction = require('./transactions/VerifiableTransaction');

var _VerifiableTransaction2 = _interopRequireDefault(_VerifiableTransaction);

var _VerificableTransactionBuilder = require('./transactions/VerificableTransactionBuilder');

var _VerificableTransactionBuilder2 = _interopRequireDefault(_VerificableTransactionBuilder);

var _keyPair = require('./crypto/keyPair');

var KeyPair = _interopRequireWildcard(_keyPair);

var _crypto = require('./crypto/crypto');

var _crypto2 = _interopRequireDefault(_crypto);

var _nacl_catapult = require('./crypto/nacl_catapult');

var _nacl_catapult2 = _interopRequireDefault(_nacl_catapult);

var _sha3Hasher = require('./crypto/sha3Hasher');

var _sha3Hasher2 = _interopRequireDefault(_sha3Hasher);

var _convert = require('./coders/convert');

var _convert2 = _interopRequireDefault(_convert);

var _idGenerator = require('./coders/idGenerator');

var _idGenerator2 = _interopRequireDefault(_idGenerator);

var _uint = require('./coders/uint64');

var _uint2 = _interopRequireDefault(_uint);

var _address = require('./coders/address');

var _address2 = _interopRequireDefault(_address);

var _NamespaceMosaicId = require('./transactions/NamespaceMosaicId');

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * This_document_defines_all_the_nis2_api_routes_and_behaviour.<br>
 * The <code>index</code> module provides access to constructors for all the classes which comprise the public API.
 * <p>
 * An AMD (recommended!) or CommonJS application will generally do something equivalent to the following:
 * <pre>
 * var Nis2Api = require('index'); // See note below*.
 * var xxxSvc = new Nis2Api.XxxApi(); // Allocate the API class we're going to use.
 * var yyyModel = new Nis2Api.Yyy(); // Construct a model instance.
 * yyyModel.someProperty = 'someValue';
 * ...
 * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
 * ...
 * </pre>
 * <em>*NOTE: For a top-level AMD script, use require(['index'], function(){...})
 * and put the application logic within the callback function.</em>
 * </p>
 * <p>
 * A non-AMD browser application (discouraged) might do something like this:
 * <pre>
 * var xxxSvc = new Nis2Api.XxxApi(); // Allocate the API class we're going to use.
 * var yyy = new Nis2Api.Yyy(); // Construct a model instance.
 * yyyModel.someProperty = 'someValue';
 * ...
 * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
 * ...
 * </pre>
 * </p>
 * @module index
 * @version 1.0.0
 */
exports.ApiClient = _ApiClient2.default;
exports.AccountDTO = _AccountDTO2.default;
exports.AccountIds = _AccountIds2.default;
exports.AccountInfoDTO = _AccountInfoDTO2.default;
exports.AccountMetaDTO = _AccountMetaDTO2.default;
exports.AnnounceTransactionInfoDTO = _AnnounceTransactionInfoDTO2.default;
exports.BlockDTO = _BlockDTO2.default;
exports.BlockInfoDTO = _BlockInfoDTO2.default;
exports.BlockMetaDTO = _BlockMetaDTO2.default;
exports.BlockchainScoreDTO = _BlockchainScoreDTO2.default;
exports.BlockchainStorageInfoDTO = _BlockchainStorageInfoDTO2.default;
exports.HeightDTO = _HeightDTO2.default;
exports.MosaicDTO = _MosaicDTO2.default;
exports.MosaicDefinitionDTO = _MosaicDefinitionDTO2.default;
exports.MosaicIds = _MosaicIds2.default;
exports.MosaicInfoDTO = _MosaicInfoDTO2.default;
exports.MosaicNameDTO = _MosaicNameDTO2.default;
exports.MosaicPropertiesDTO = _MosaicPropertiesDTO2.default;
exports.Multisig = _Multisig2.default;
exports.MultisigAccountGraphInfoDTO = _MultisigAccountGraphInfoDTO2.default;
exports.MultisigAccountInfoDTO = _MultisigAccountInfoDTO2.default;
exports.NamespaceDTO = _NamespaceDTO2.default;
exports.NamespaceIds = _NamespaceIds2.default;
exports.NamespaceInfoDTO = _NamespaceInfoDTO2.default;
exports.NamespaceMosaicMetaDTO = _NamespaceMosaicMetaDTO2.default;
exports.NamespaceNameDTO = _NamespaceNameDTO2.default;
exports.TransactionHashes = _TransactionHashes2.default;
exports.TransactionIds = _TransactionIds2.default;
exports.TransactionPayload = _TransactionPayload2.default;
exports.TransactionStatusDTO = _TransactionStatusDTO2.default;
exports.UInt64DTO = _UInt64DTO2.default;
exports.AccountRoutesApi = _AccountRoutesApi2.default;
exports.BlockchainRoutesApi = _BlockchainRoutesApi2.default;
exports.MosaicRoutesApi = _MosaicRoutesApi2.default;
exports.NamespaceRoutesApi = _NamespaceRoutesApi2.default;
exports.TransactionRoutesApi = _TransactionRoutesApi2.default;
exports.NetworkRoutesApi = _NetworkRoutesApi2.default;
exports.BlockListener = _BlockListener2.default;
exports.ConfirmedTransactionsListener = _ConfirmedTransactionsListener2.default;
exports.PartialTransactionsListener = _PartialTransactionsListener2.default;
exports.TransactionStatusListener = _TransactionStatusListener2.default;
exports.UnconfirmedTransactionsListener = _UnconfirmedTransactionsListener2.default;
exports.Schema = _Schema2.default;
exports.CosignatureTransaction = _CosignatureTransaction2.default;
exports.AggregateTransaction = _AggregateTransaction2.default;
exports.deadline = _Deadline2.default;
exports.HashLockTransaction = _HashLockTransaction2.default;
exports.MosaicCreationTransaction = _MosaicCreationTransaction2.default;
exports.MosaicSupplyChangeTransaction = _MosaicSupplyChangeTransaction2.default;
exports.MultisigModificationTransaction = _MultisigModificationTransaction2.default;
exports.NamespaceCreationTransaction = _NamespaceCreationTransaction2.default;
exports.SecretLockTransaction = _SecretLockTransaction2.default;
exports.SecretProofTransaction = _SecretProofTransaction2.default;
exports.TransferTransaction = _TransferTransaction2.default;
exports.VerifiableTransaction = _VerifiableTransaction2.default;
exports.VerifiableTransactionBuilder = _VerificableTransactionBuilder2.default;
exports.KeyPair = KeyPair;
exports.crypto = _crypto2.default;
exports.nacl_catapult = _nacl_catapult2.default;
exports.convert = _convert2.default;
exports.sha3Hasher = _sha3Hasher2.default;
exports.IdGenerator = _idGenerator2.default;
exports.uint64 = _uint2.default;
exports.address = _address2.default;
exports.mosaicId = _NamespaceMosaicId.mosaicId;
exports.namespaceId = _NamespaceMosaicId.namespaceId;
exports.subnamespaceParentId = _NamespaceMosaicId.subnamespaceParentId;
exports.subnamespaceNamespaceId = _NamespaceMosaicId.subnamespaceNamespaceId; /*
                                                                               * Copyright 2018 NEM
                                                                               *
                                                                               * Licensed under the Apache License, Version 2.0 (the "License");
                                                                               * you may not use this file except in compliance with the License.
                                                                               * You may obtain a copy of the License at
                                                                               *
                                                                               *     http://www.apache.org/licenses/LICENSE-2.0
                                                                               *
                                                                               * Unless required by applicable law or agreed to in writing, software
                                                                               * distributed under the License is distributed on an "AS IS" BASIS,
                                                                               * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
                                                                               * See the License for the specific language governing permissions and
                                                                               * limitations under the License.
                                                                               */

/**
 * NIS2 API
 * This document defines all the nis2 api routes and behaviour
 *
 * OpenAPI spec version: 1.0.0
 * Contact: guillemchain@gmail.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */
