// nem2-sdk/dist/src/service/MosaicService.js
"use strict";
/*
 * Copyright 2018 NEM
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
require("rxjs/add/observable/of");
require("rxjs/add/operator/catch");
require("rxjs/add/operator/filter");
require("rxjs/add/operator/mergeMap");
require("rxjs/add/operator/toArray");
const Observable_1 = require("rxjs/Observable");
const MosaicAmountView_1 = require("./MosaicAmountView");
const MosaicView_1 = require("./MosaicView");
/**
 * Mosaic service
 */
class MosaicService {
    /**
     * Constructor
     * @param accountHttp
     * @param mosaicHttp
     * @param namespaceHttp
     */
    constructor(accountHttp, mosaicHttp, namespaceHttp) {
        this.accountHttp = accountHttp;
        this.mosaicHttp = mosaicHttp;
        this.namespaceHttp = namespaceHttp;
    }
    /**
     * Get mosaic view given mosaicIds
     * @param mosaicIds - The ids of the mosaics
     * @returns {Observable<MosaicView[]>}
     */
    mosaicsView(mosaicIds) {
        return Observable_1.Observable.of(mosaicIds)
            .flatMap((_) => this.mosaicHttp.getMosaics(mosaicIds))
            .flatMap((_) => _)
            .flatMap((mosaicInfo) => this.mosaicHttp.getMosaicsName([mosaicInfo.mosaicId]).map((mosaicsName) => {
            return { mosaicInfo, mosaicName: mosaicsName[0].name };
        }))
            .flatMap((_) => this.namespaceHttp.getNamespacesName([_.mosaicInfo.namespaceId]).map((namespacesName) => {
            return new MosaicView_1.MosaicView(_.mosaicInfo, namespacesName[0].name, _.mosaicName);
        }))
            .toArray();
    }
    /**
     * Get mosaic amount view given mosaic array
     * @param mosaics
     * @returns {Observable<MosaicAmountView[]>}
     */
    mosaicsAmountView(mosaics) {
        return Observable_1.Observable.of(mosaics)
            .flatMap((_) => _)
            .flatMap((mosaic) => this.mosaicsView([mosaic.id])
            .filter((_) => _.length !== 0)
            .map((mosaicViews) => {
            return new MosaicAmountView_1.MosaicAmountView(mosaicViews[0].mosaicInfo, mosaicViews[0].namespaceName, mosaicViews[0].mosaicName, mosaic.amount);
        }))
            .toArray();
    }
    /**
     * Get balance mosaics in form of MosaicAmountViews for a given account address
     * @param address - Account address
     * @returns {Observable<MosaicAmountView[]>}
     */
    mosaicsAmountViewFromAddress(address) {
        return Observable_1.Observable.of(address)
            .flatMap((_) => this.accountHttp.getAccountInfo(_))
            .flatMap((_) => this.mosaicsAmountView(_.mosaics));
    }
}
exports.MosaicService = MosaicService;
