// nem2-library/dist/api/AccountRoutesApi.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /*
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Copyright 2018 NEM
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Licensed under the Apache License, Version 2.0 (the "License");
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * you may not use this file except in compliance with the License.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * You may obtain a copy of the License at
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *     http://www.apache.org/licenses/LICENSE-2.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Unless required by applicable law or agreed to in writing, software
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * distributed under the License is distributed on an "AS IS" BASIS,
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * See the License for the specific language governing permissions and
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * limitations under the License.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

/**
 * NIS2 API
 * This document defines all the nis2 api routes and behaviour
 *
 * OpenAPI spec version: 1.0.0
 * Contact: guillemchain@gmail.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _AccountInfoDTO = require('../model/AccountInfoDTO');

var _AccountInfoDTO2 = _interopRequireDefault(_AccountInfoDTO);

var _Addresses = require('../model/Addresses');

var _Addresses2 = _interopRequireDefault(_Addresses);

var _MultisigAccountGraphInfoDTO = require('../model/MultisigAccountGraphInfoDTO');

var _MultisigAccountGraphInfoDTO2 = _interopRequireDefault(_MultisigAccountGraphInfoDTO);

var _MultisigAccountInfoDTO = require('../model/MultisigAccountInfoDTO');

var _MultisigAccountInfoDTO2 = _interopRequireDefault(_MultisigAccountInfoDTO);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* AccountRoutes service.
* @module api/AccountRoutesApi
* @version 1.0.0
*/
var AccountRoutesApi = function () {

  /**
  * Constructs a new AccountRoutesApi. 
  * @alias module:api/AccountRoutesApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function AccountRoutesApi(apiClient) {
    _classCallCheck(this, AccountRoutesApi);

    this.apiClient = apiClient || _ApiClient2.default.instance;
  }

  /**
   * Get account information
   * Returns account information for a given address or publicKey
   * @param {String} accountId Account address or publicKey
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/AccountInfoDTO} and HTTP response
   */


  _createClass(AccountRoutesApi, [{
    key: 'getAccountInfoWithHttpInfo',
    value: function getAccountInfoWithHttpInfo(accountId) {
      var postBody = null;

      // verify the required parameter 'accountId' is set
      if (accountId === undefined || accountId === null) {
        throw new Error("Missing the required parameter 'accountId' when calling getAccountInfo");
      }

      var pathParams = {
        'accountId': accountId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};

      var authNames = [];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _AccountInfoDTO2.default;

      return this.apiClient.callApi('/account/{accountId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType);
    }

    /**
     * Get account information
     * Returns account information for a given address or publicKey
     * @param {String} accountId Account address or publicKey
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/AccountInfoDTO}
     */

  }, {
    key: 'getAccountInfo',
    value: function getAccountInfo(accountId) {
      return this.getAccountInfoWithHttpInfo(accountId).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get account information
     * Returns information for a given address or publicKey
     * @param {String} accountId The multisig account address or public key for which information should be retreived.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/MultisigAccountInfoDTO} and HTTP response
     */

  }, {
    key: 'getAccountMultisigWithHttpInfo',
    value: function getAccountMultisigWithHttpInfo(accountId) {
      var postBody = null;

      // verify the required parameter 'accountId' is set
      if (accountId === undefined || accountId === null) {
        throw new Error("Missing the required parameter 'accountId' when calling getAccountMultisig");
      }

      var pathParams = {
        'accountId': accountId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};

      var authNames = [];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _MultisigAccountInfoDTO2.default;

      return this.apiClient.callApi('/account/{accountId}/multisig', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType);
    }

    /**
     * Get account information
     * Returns information for a given address or publicKey
     * @param {String} accountId The multisig account address or public key for which information should be retreived.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/MultisigAccountInfoDTO}
     */

  }, {
    key: 'getAccountMultisig',
    value: function getAccountMultisig(accountId) {
      return this.getAccountMultisigWithHttpInfo(accountId).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get multisig graph information
     * Returns information of all accounts address or publicKey is cosignatory
     * @param {String} accountId The multisig account address public key for which information should be retreived.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link Array.<module:model/MultisigAccountGraphInfoDTO>} and HTTP response
     */

  }, {
    key: 'getAccountMultisigGraphWithHttpInfo',
    value: function getAccountMultisigGraphWithHttpInfo(accountId) {
      var postBody = null;

      // verify the required parameter 'accountId' is set
      if (accountId === undefined || accountId === null) {
        throw new Error("Missing the required parameter 'accountId' when calling getAccountMultisigGraph");
      }

      var pathParams = {
        'accountId': accountId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};

      var authNames = [];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = [_MultisigAccountGraphInfoDTO2.default];

      return this.apiClient.callApi('/account/{accountId}/multisig/graph', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType);
    }

    /**
     * Get multisig graph information
     * Returns information of all accounts address or publicKey is cosignatory
     * @param {String} accountId The multisig account address public key for which information should be retreived.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link Array.<module:model/MultisigAccountGraphInfoDTO>}
     */

  }, {
    key: 'getAccountMultisigGraph',
    value: function getAccountMultisigGraph(accountId) {
      return this.getAccountMultisigGraphWithHttpInfo(accountId).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get accounts information
     * Returns accounts information for a given set of address and publicKey
     * @param {module:model/Addresses} addresses Accounts address array
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link Array.<module:model/AccountInfoDTO>} and HTTP response
     */

  }, {
    key: 'getAccountsInfoWithHttpInfo',
    value: function getAccountsInfoWithHttpInfo(addresses) {
      var postBody = addresses;

      // verify the required parameter 'addresses' is set
      if (addresses === undefined || addresses === null) {
        throw new Error("Missing the required parameter 'addresses' when calling getAccountsInfo");
      }

      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};

      var authNames = [];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = [_AccountInfoDTO2.default];

      return this.apiClient.callApi('/account', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType);
    }

    /**
     * Get accounts information
     * Returns accounts information for a given set of address and publicKey
     * @param {module:model/Addresses} addresses Accounts address array
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link Array.<module:model/AccountInfoDTO>}
     */

  }, {
    key: 'getAccountsInfo',
    value: function getAccountsInfo(addresses) {
      return this.getAccountsInfoWithHttpInfo(addresses).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get incoming transactions information
     * Returns incoming transactions information for a given publicKey
     * @param {String} publicKey Account publicKey
     * @param {Object} opts Optional parameters
     * @param {Number} opts.pageSize The numbers of transactions to return
     * @param {String} opts.id Id last transaction id to apply pagination
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link Array.<Object>} and HTTP response
     */

  }, {
    key: 'incomingTransactionsWithHttpInfo',
    value: function incomingTransactionsWithHttpInfo(publicKey, opts) {
      opts = opts || {};
      var postBody = null;

      // verify the required parameter 'publicKey' is set
      if (publicKey === undefined || publicKey === null) {
        throw new Error("Missing the required parameter 'publicKey' when calling incomingTransactions");
      }

      var pathParams = {
        'publicKey': publicKey
      };
      var queryParams = {
        'pageSize': opts['pageSize'],
        'id': opts['id']
      };
      var headerParams = {};
      var formParams = {};

      var authNames = [];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = [Object];

      return this.apiClient.callApi('/account/{publicKey}/transactions/incoming', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType);
    }

    /**
     * Get incoming transactions information
     * Returns incoming transactions information for a given publicKey
     * @param {String} publicKey Account publicKey
     * @param {Object} opts Optional parameters
     * @param {Number} opts.pageSize The numbers of transactions to return
     * @param {String} opts.id Id last transaction id to apply pagination
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link Array.<Object>}
     */

  }, {
    key: 'incomingTransactions',
    value: function incomingTransactions(publicKey, opts) {
      return this.incomingTransactionsWithHttpInfo(publicKey, opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get outgoing transactions information
     * Returns outgoing transactions information for a given publicKey
     * @param {String} publicKey Account publicKey
     * @param {Object} opts Optional parameters
     * @param {Number} opts.pageSize The numbers of transactions to return
     * @param {String} opts.id Id last transaction id to apply pagination
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link Array.<Object>} and HTTP response
     */

  }, {
    key: 'outgoingTransactionsWithHttpInfo',
    value: function outgoingTransactionsWithHttpInfo(publicKey, opts) {
      opts = opts || {};
      var postBody = null;

      // verify the required parameter 'publicKey' is set
      if (publicKey === undefined || publicKey === null) {
        throw new Error("Missing the required parameter 'publicKey' when calling outgoingTransactions");
      }

      var pathParams = {
        'publicKey': publicKey
      };
      var queryParams = {
        'pageSize': opts['pageSize'],
        'id': opts['id']
      };
      var headerParams = {};
      var formParams = {};

      var authNames = [];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = [Object];

      return this.apiClient.callApi('/account/{publicKey}/transactions/outgoing', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType);
    }

    /**
     * Get outgoing transactions information
     * Returns outgoing transactions information for a given publicKey
     * @param {String} publicKey Account publicKey
     * @param {Object} opts Optional parameters
     * @param {Number} opts.pageSize The numbers of transactions to return
     * @param {String} opts.id Id last transaction id to apply pagination
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link Array.<Object>}
     */

  }, {
    key: 'outgoingTransactions',
    value: function outgoingTransactions(publicKey, opts) {
      return this.outgoingTransactionsWithHttpInfo(publicKey, opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get partial transactions information
     * Returns partial transactions information for a given publicKey
     * @param {String} publicKey Account publicKey
     * @param {Object} opts Optional parameters
     * @param {Number} opts.pageSize The numbers of transactions to return
     * @param {String} opts.id Id last transaction id to apply pagination
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link Array.<Object>} and HTTP response
     */

  }, {
    key: 'partialTransactionsWithHttpInfo',
    value: function partialTransactionsWithHttpInfo(publicKey, opts) {
      opts = opts || {};
      var postBody = null;

      // verify the required parameter 'publicKey' is set
      if (publicKey === undefined || publicKey === null) {
        throw new Error("Missing the required parameter 'publicKey' when calling partialTransactions");
      }

      var pathParams = {
        'publicKey': publicKey
      };
      var queryParams = {
        'pageSize': opts['pageSize'],
        'id': opts['id']
      };
      var headerParams = {};
      var formParams = {};

      var authNames = [];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = [Object];

      return this.apiClient.callApi('/account/{publicKey}/transactions/partial', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType);
    }

    /**
     * Get partial transactions information
     * Returns partial transactions information for a given publicKey
     * @param {String} publicKey Account publicKey
     * @param {Object} opts Optional parameters
     * @param {Number} opts.pageSize The numbers of transactions to return
     * @param {String} opts.id Id last transaction id to apply pagination
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link Array.<Object>}
     */

  }, {
    key: 'partialTransactions',
    value: function partialTransactions(publicKey, opts) {
      return this.partialTransactionsWithHttpInfo(publicKey, opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get transactions information
     * Returns transactions information for a given publicKey
     * @param {String} publicKey Account publicKey
     * @param {Object} opts Optional parameters
     * @param {Number} opts.pageSize The numbers of transactions to return
     * @param {String} opts.id Id last transaction id to apply pagination
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link Array.<Object>} and HTTP response
     */

  }, {
    key: 'transactionsWithHttpInfo',
    value: function transactionsWithHttpInfo(publicKey, opts) {
      opts = opts || {};
      var postBody = null;

      // verify the required parameter 'publicKey' is set
      if (publicKey === undefined || publicKey === null) {
        throw new Error("Missing the required parameter 'publicKey' when calling transactions");
      }

      var pathParams = {
        'publicKey': publicKey
      };
      var queryParams = {
        'pageSize': opts['pageSize'],
        'id': opts['id']
      };
      var headerParams = {};
      var formParams = {};

      var authNames = [];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = [Object];

      return this.apiClient.callApi('/account/{publicKey}/transactions', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType);
    }

    /**
     * Get transactions information
     * Returns transactions information for a given publicKey
     * @param {String} publicKey Account publicKey
     * @param {Object} opts Optional parameters
     * @param {Number} opts.pageSize The numbers of transactions to return
     * @param {String} opts.id Id last transaction id to apply pagination
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link Array.<Object>}
     */

  }, {
    key: 'transactions',
    value: function transactions(publicKey, opts) {
      return this.transactionsWithHttpInfo(publicKey, opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get unconfirmed transactions information
     * Returns unconfirmed transactions information for a given publicKey
     * @param {String} publicKey Account publicKey
     * @param {Object} opts Optional parameters
     * @param {Number} opts.pageSize The numbers of transactions to return
     * @param {String} opts.id Id last transaction id to apply pagination
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link Array.<Object>} and HTTP response
     */

  }, {
    key: 'unconfirmedTransactionsWithHttpInfo',
    value: function unconfirmedTransactionsWithHttpInfo(publicKey, opts) {
      opts = opts || {};
      var postBody = null;

      // verify the required parameter 'publicKey' is set
      if (publicKey === undefined || publicKey === null) {
        throw new Error("Missing the required parameter 'publicKey' when calling unconfirmedTransactions");
      }

      var pathParams = {
        'publicKey': publicKey
      };
      var queryParams = {
        'pageSize': opts['pageSize'],
        'id': opts['id']
      };
      var headerParams = {};
      var formParams = {};

      var authNames = [];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = [Object];

      return this.apiClient.callApi('/account/{publicKey}/transactions/unconfirmed', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType);
    }

    /**
     * Get unconfirmed transactions information
     * Returns unconfirmed transactions information for a given publicKey
     * @param {String} publicKey Account publicKey
     * @param {Object} opts Optional parameters
     * @param {Number} opts.pageSize The numbers of transactions to return
     * @param {String} opts.id Id last transaction id to apply pagination
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link Array.<Object>}
     */

  }, {
    key: 'unconfirmedTransactions',
    value: function unconfirmedTransactions(publicKey, opts) {
      return this.unconfirmedTransactionsWithHttpInfo(publicKey, opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }
  }]);

  return AccountRoutesApi;
}();

exports.default = AccountRoutesApi;
